src/backend/distributed/executor/local_executor.c

```diff
@@ -206,7 +206,7 @@ ExecuteLocalTaskList(CitusScanState *scanState, List *taskList)
 			localPlan = planner(shardQuery, cursorOptions, paramListInfo);
 		}
 
-		LogLocalCommand(task);
+		LogLocalCommand(TaskQueryString(task));
```

Comment:
Why are we calling TaskQueryString in this code path? What if we don't need the query string?

Reply:
Maybe revert this change and add a comment to LogLocalCommand that we're passing the Task rather than the query string to avoid computing it unnecessarily.

Reply:
Right it would cause unnecessary operation. 

I think we should separate this into two things then, `ShouldLogLocalCommand` and `LogLocalCommand`. Honestly to me, the current (master) implementation seems a little bit strange, because it is possible that we wont log the command even though the function name says `LogLocalCommand`. I would expect the message to be logged if I am calling that function. So separating this into two methods makes more sense to me